### PR TITLE
Fix contract key uniqueness check in kvutils

### PR DESCRIPTION
**Pull request by dajmaki**
Archival of a contract with a key and recreation within the
same transaction is now allowed in kvutils.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [x] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/ContractKeys.scala [hunk 1]
```diff
@@ -162,10 +163,22 @@ final class ContractKeys(session: LedgerSession) extends LedgerTestSuite(session
       }
     }
 
+  val recreateContractKeys =
+    LedgerTest("CKRecreate", "Contract keys can be recreated in single transaction") { context =>
+      val key = s"${UUID.randomUUID.toString}-key"
+      for {
+        ledger <- context.participant()
+        Vector(owner) <- ledger.allocateParties(1)
```
Comment (gerolf-da) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/ContractKeys.scala]: You can use `ledger.allocateParty()` to only get back a single one.

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/ContractKeys.scala [hunk 2]
```diff
@@ -162,10 +163,22 @@ final class ContractKeys(session: LedgerSession) extends LedgerTestSuite(session
...
+        delegated <- ledger.create(owner, Delegated(owner, key))
+        _ <- ledger.exercise(owner, delegated.exerciseRecreate(_))
+      } yield { () }
```
Comment (gerolf-da) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/ContractKeys.scala]: I think we can still make some assertions here:
for example:
asserting that the contract key from the original `Delegated` contract is the same as the contract key from the replacement, but the contract IDs themselves are different.

Comment (dajmaki) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/ContractKeys.scala]: thanks, good point. will add the assertions.

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/ContractKeys.scala (hunk 1 above)
Comment (stefanobaghino-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/ContractKeys.scala]: Which would be:
```suggestion
        owner <- ledger.allocateParty()
```

Comment (stefanobaghino-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/ContractKeys.scala]: Ah, already done, sorry, just noticed.

**Pull request merged**